### ISSUE TITLE
Automate python version management?

### ISSUE THREAD

**Issue by pcattori** [kennethreitz/pipenv#530]
`Pipfile`s have a `[require]` section where `python_version` can be specified.

Currently AFAIK it is up to the user to manually install that version of Python on their machine and pass it to `pipenv` via the `--python` flag.

I think its desirable to:
(1) automatically download/install a new Python version if there are no existing Python versions compatible with the project already on the machine
(2) specify Python version number instead of executable

eg.
```sh
$ pipenv
This project requires python version '>=3.6'. No compatible python version found.
Downloading python 3.6.2...
Installing python 3.6.2...
<rest of project dependency installation happens here>
...
```

```sh
$ pipenv init --python >=2.7.8,==2.*
Compatible python version found! Using python 2.7.12.
<rest of project initialization happens here>
...
```

This could be done by integrating with `pyenv`. Here's a [proof-of-concept](https://github.com/pcattori/snek/blob/master/snek/cli.py#L68) for the `pyenv` integration.

In short, python is just another dependency of the project that is explicitly specified in the `Pipfile` and should be treated as such.

Would this functionality be desirable for `pipenv`?

**Comment by kennethreitz** [comment]
Maybe. Maybe not.

**Comment by kennethreitz** [comment]
I'll play with it.

**Comment by kennethreitz** [comment]
@pcattori we can't use fstrings, as they're incompatible with python 2.

**Comment by kennethreitz** [comment]
pyenv doesn't install on my machine — looks like they're having problems. 

we already have a dependency on `pythonz`, so might make sense to go off that.

**Comment by kennethreitz** [comment]
not doing this. but, got one step closer

**Comment by kennethreitz** [comment]
https://github.com/kennethreitz/pipenv/pull/532

**Comment by webknjaz** [comment]
@kennethreitz I've installed pipenv in my pyenv environment and it's got broken after `pipenv update` (I think). pipenv pretends to install stuff (and it looks it does), but the virtualenv it uses isn't activated automatically.

**Comment by kennethreitz** [comment]
we never automatically activate your virtualenv. that's what `pipenv shell` is for.

**Comment by webknjaz** [comment]
@kennethreitz that's what I meant: `pipenv shell` thinks it's already in virtualenv and does nothing while commands installing packages actually spawn `pip install` in some virtualenv (confirmed with `--verbose`)
